parser-typechecker/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -1280,28 +1291,29 @@ showDiffNamespace ppe d@OBD.BranchDiffOutput{..} =
                      -> [OBD.RenameTermDisplay v a]
                      -> Numbered [Pretty]
   prettyRenameGroups types terms =
-    (<>) <$> traverse prettyGroup types
+    (<>) <$> traverse (prettyGroup . (over _1 Referent.Ref)) types
          <*> traverse prettyGroup terms
     where
         leftNamePad :: Int = foldl1' max $
           map (foldl1' max . map HQ'.nameLength . toList . view _3) terms <>
           map (foldl1' max . map HQ'.nameLength . toList . view _3) types
-        prettyGroup :: (a, b, Set HQ'.HashQualified, Set HQ'.HashQualified)
+        prettyGroup :: (Referent, b, Set HQ'.HashQualified, Set HQ'.HashQualified)
                     -> Numbered Pretty
-        prettyGroup (_, _, olds, news) = let
+        prettyGroup (r, _, olds, news) = let
           -- [ "peach  ┐"
           -- , "peach' ┘"]
           olds' :: [Numbered Pretty] =
-            map (\old -> num <&> (\n -> n <> old))
-              . P.boxRight' P.rBoxStyle2
+            map (\(oldhq, oldp) -> numHQ oldPath oldhq r <&> (\n -> n <> oldp))
+              . (zip (toList olds))
+              . (P.boxRight' P.rBoxStyle2)
```

Comment:
Instead of just generating a new number with `num`, we number and create a `numberedArgs` entry in a single step with `numHQ`.